tests/framework/web/UrlManagerCreateUrlTest.php

```diff
@@ -758,4 +759,34 @@ public function testUrlCreateCacheWithParameterMismatch()
         $this->assertEquals(2, $rules[0]->createCounter);
         $this->assertEquals(1, $rules[1]->createCounter);
     }
+
+    /**
+     * @see https://github.com/yiisoft/yii2/issues/14406
+     */
+    public function testCreatingRulesWithDifferentRuleConfigAndEnabledCache()
+    {
+        $this->mockWebApplication([
+            'components' => [
+                'cache' => ArrayCache::className()
+            ]
+        ]);
+        $urlManager = $this->getUrlManager([
```

Comment:
$urlManager does not seem to be used and is therefore hard to understand. Perhabs can you add an assertion similar to what you  do for $cachedUrlManager